test/node/index.js

```diff
@@ -116,6 +116,49 @@ describe('parser', function() {
     });
   });
 
+  it('should decode an ArrayBuffer/Buffer with undeclared type as binary', function(done) {
+    var buffer1 = new ArrayBuffer(4);
+    var dataview = new DataView(buffer1);
+    var buffer2 = new Buffer(4);
+    for (var i = 0; i < buffer1.byteLength ; i++) {
+      dataview.setInt8(i, 4-i);
+      buffer2[i] = 4-i;
+    }
+
+    var decoded1 = decode(buffer1, null);
+    var decoded2 = decode(buffer2, null);
+    expect(decoded1).to.eql({ type: 'message', data: new Buffer([3,2,1])});
+    expect(new Uint8Array(decoded1.data)).to.eql(new Uint8Array(buffer1.slice(1)));
+    expect(decoded2).to.eql({ type: 'message', data: new Buffer(buffer2.slice(1))});
+    expect(new Uint8Array(decoded2.data)).to.eql(new Uint8Array(buffer2.slice(1)));
+    done();
+  });
+
+  it('should decode an ArrayBuffer/Buffer as binary of type ArrayBuffer', function(done) {
```

Comment:
My concern is that some of the tests you added may already exist elsewhere in the file.

I agree though that the output of both method `encode` and `decode` could be tested, but that seems out of the scope of that particular PR, doesn't it?